dns_records.tf

```diff
@@ -0,0 +1,316 @@
+#-------------------------------------------------------------------------------
+# Create all necessary DNS records for the IPA servers.
+# -------------------------------------------------------------------------------
+locals {
+  # The Route53 zone where the IPA server DNS records should be
+  # created.
+  zone_id = data.terraform_remote_state.networking.outputs.private_zone.id
+
+  # Route53 reverse zones corresponding to the subnets where the
+  # FreeIPA servers reside.
+  reverse_zone_ids = [for zone in data.terraform_remote_state.networking.outputs.private_subnet_private_reverse_zones : zone.id]
+}
+
+resource "aws_route53_record" "server0_A" {
```

Comment:
I don't like to use `count` because it sometimes leads to an ordering problem, where resources get recreated just because they are now `1` and they used to be `0`.  I prefer to use a `map` with keys that tie to the actual resource.

Please see commits d2eafb6 and f32a936.

Reply:
That's fair. It just scratched at my DRY urges. I appreciate the consolidation!